doc/manual/src/language/string-interpolation.md

```diff
@@ -0,0 +1,97 @@
+# String interpolation
+
+String interpolation is a language feature where a [string], [path], or [attribute name] can contain expressions enclosed in `${ }` (dollar-sign with curly brackets).
+
+Such a string is an *interpolated string*, and an expression inside is an *interpolated expression*.
+
+Interpolated expressions must evaluate to something that can be coerced to a string:
```

Comment:
I'd add `, such as:` here at the end because the list is not exhaustive.

Reply:
I'd like to make the list exhaustive. What else is missing?

Reply:
Most important additional thing are attribute sets with the `__toString` attribute (which is a function `self -> string`):

```
nix-repl> "${{ __toString = _: "toString"; }}"
"toString"
```

This is sometimes used to make some kind of "smart value" that has e.g. `__toString` and `__functor` to pretend to be an object or whatever.

Technically saying that `derivations` are coercible to string is also not precise, as it is all attribute sets with an `outPath` attribute:

```
nix-repl> "${{ outPath = "outPath"; }}"        
"outPath"
```

This is the behaviour for what we call "weak coercion" in Tvix (equivalent to `!coerceMore` in C++ Nix), there's slightly different behaviour for "strong coercion" but I'm pretty sure it doesn't apply to interpolation.

Reply:
Yes, of course. Now I remember why I didn't put it there: I wasn't sure if we want to expose this feature.

This PR was originally only to rename antiquotation to string interpolation.

In the Nix team meeting we agreed to add a glossary entry to have something to point to.

Then @edolstra asked to extract that into its own section.

I tried to keep changes minimal and preserve the contents as far as possible.

Describing what a derivation is and how it works specifically is a completely different scope, and the changes here are already a bit too much to review easily.

I'd rather postpone going into the `__toString` business to a different PR.